<!DOCTYPE html>
<html>
<head><title>R: Single-Cell Analysis Toolkit for Gene Expression Data in R</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">
<h1> Single-Cell Analysis Toolkit for Gene Expression Data in R
<img class="toplogo" src="../../../doc/html/Rlogo.svg" alt="[R logo]" />
</h1>
<hr/>
<div style="text-align: center;">
<a href="../../../doc/html/packages.html"><img class="arrow" src="../../../doc/html/left.jpg" alt="[Up]" /></a>
<a href="../../../doc/html/index.html"><img class="arrow" src="../../../doc/html/up.jpg" alt="[Top]" /></a>
</div><h2>Documentation for package &lsquo;scater&rsquo; version 1.26.1</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
<li><a href="../doc/index.html">User guides, package vignettes and other documentation.</a></li>
<li><a href="../NEWS">Package NEWS</a>.</li>
</ul>

<h2>Help Pages</h2>


<table style="width: 100%;">
<tr><td style="width: 25%;"><a href="annotateBMFeatures.html">annotateBMFeatures</a></td>
<td>Get feature annotation information from Biomart</td></tr>
<tr><td style="width: 25%;"><a href="bootstraps.html">bootstraps</a></td>
<td>Accessor and replacement for bootstrap results in a 'SingleCellExperiment' object</td></tr>
<tr><td style="width: 25%;"><a href="bootstraps.html">bootstraps-method</a></td>
<td>Accessor and replacement for bootstrap results in a 'SingleCellExperiment' object</td></tr>
<tr><td style="width: 25%;"><a href="bootstraps.html">bootstraps&lt;-</a></td>
<td>Accessor and replacement for bootstrap results in a 'SingleCellExperiment' object</td></tr>
<tr><td style="width: 25%;"><a href="bootstraps.html">bootstraps&lt;--method</a></td>
<td>Accessor and replacement for bootstrap results in a 'SingleCellExperiment' object</td></tr>
<tr><td style="width: 25%;"><a href="defunct.html">calculateDiffusionMap</a></td>
<td>Defunct functions</td></tr>
<tr><td style="width: 25%;"><a href="defunct.html">calculateDiffusionMap-method</a></td>
<td>Defunct functions</td></tr>
<tr><td style="width: 25%;"><a href="runMDS.html">calculateMDS</a></td>
<td>Perform MDS on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="runMDS.html">calculateMDS-method</a></td>
<td>Perform MDS on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="runMultiUMAP.html">calculateMultiUMAP</a></td>
<td>Multi-modal UMAP</td></tr>
<tr><td style="width: 25%;"><a href="runMultiUMAP.html">calculateMultiUMAP-method</a></td>
<td>Multi-modal UMAP</td></tr>
<tr><td style="width: 25%;"><a href="runNMF.html">calculateNMF</a></td>
<td>Perform NMF on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="runNMF.html">calculateNMF-method</a></td>
<td>Perform NMF on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="runPCA.html">calculatePCA</a></td>
<td>Perform PCA on expression data</td></tr>
<tr><td style="width: 25%;"><a href="runPCA.html">calculatePCA-method</a></td>
<td>Perform PCA on expression data</td></tr>
<tr><td style="width: 25%;"><a href="defunct.html">calculateQCMetrics</a></td>
<td>Defunct functions</td></tr>
<tr><td style="width: 25%;"><a href="runTSNE.html">calculateTSNE</a></td>
<td>Perform t-SNE on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="runTSNE.html">calculateTSNE-method</a></td>
<td>Perform t-SNE on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="runUMAP.html">calculateUMAP</a></td>
<td>Perform UMAP on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="runUMAP.html">calculateUMAP-method</a></td>
<td>Perform UMAP on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="defunct.html">centreSizeFactors</a></td>
<td>Defunct functions</td></tr>
<tr><td style="width: 25%;"><a href="defunct.html">defunct</a></td>
<td>Defunct functions</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">exprs</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">exprs,SingleCellExperiment-method,</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">exprs&lt;--method</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">fpkm</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">fpkm-method</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">fpkm&lt;-</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">fpkm&lt;--method</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="annotateBMFeatures.html">getBMFeatureAnnos</a></td>
<td>Get feature annotation information from Biomart</td></tr>
<tr><td style="width: 25%;"><a href="getExplanatoryPCs.html">getExplanatoryPCs</a></td>
<td>Per-PC variance explained by a variable</td></tr>
<tr><td style="width: 25%;"><a href="getVarianceExplained.html">getVarianceExplained</a></td>
<td>Per-gene variance explained by a variable</td></tr>
<tr><td style="width: 25%;"><a href="getVarianceExplained.html">getVarianceExplained-method</a></td>
<td>Per-gene variance explained by a variable</td></tr>
<tr><td style="width: 25%;"><a href="ggsce.html">ggcells</a></td>
<td>Create a ggplot from a SingleCellExperiment</td></tr>
<tr><td style="width: 25%;"><a href="ggsce.html">ggfeatures</a></td>
<td>Create a ggplot from a SingleCellExperiment</td></tr>
<tr><td style="width: 25%;"><a href="multiplot.html">multiplot</a></td>
<td>Multiple plot function for ggplot2 plots</td></tr>
<tr><td style="width: 25%;"><a href="nexprs.html">nexprs</a></td>
<td>Count the number of non-zero counts per cell or feature</td></tr>
<tr><td style="width: 25%;"><a href="nexprs.html">nexprs-method</a></td>
<td>Count the number of non-zero counts per cell or feature</td></tr>
<tr><td style="width: 25%;"><a href="defunct.html">normalize-method</a></td>
<td>Defunct functions</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">norm_exprs</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">norm_exprs-method</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">norm_exprs&lt;-</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">norm_exprs&lt;--method</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="plotColData.html">plotColData</a></td>
<td>Plot column metadata</td></tr>
<tr><td style="width: 25%;"><a href="plot_reddim.html">plotDiffusionMap</a></td>
<td>Plot specific reduced dimensions</td></tr>
<tr><td style="width: 25%;"><a href="plotDots.html">plotDots</a></td>
<td>Create a dot plot of expression values</td></tr>
<tr><td style="width: 25%;"><a href="plotExplanatoryPCs.html">plotExplanatoryPCs</a></td>
<td>Plot the explanatory PCs for each variable</td></tr>
<tr><td style="width: 25%;"><a href="plotExplanatoryVariables.html">plotExplanatoryVariables</a></td>
<td>Plot explanatory variables ordered by percentage of variance explained</td></tr>
<tr><td style="width: 25%;"><a href="plotExpression.html">plotExpression</a></td>
<td>Plot expression values for all cells</td></tr>
<tr><td style="width: 25%;"><a href="plotGroupedHeatmap.html">plotGroupedHeatmap</a></td>
<td>Plot heatmap of group-level expression averages</td></tr>
<tr><td style="width: 25%;"><a href="plotHeatmap.html">plotHeatmap</a></td>
<td>Plot heatmap of gene expression values</td></tr>
<tr><td style="width: 25%;"><a href="plotHighestExprs.html">plotHighestExprs</a></td>
<td>Plot the highest expressing features</td></tr>
<tr><td style="width: 25%;"><a href="plot_reddim.html">plotMDS</a></td>
<td>Plot specific reduced dimensions</td></tr>
<tr><td style="width: 25%;"><a href="plot_reddim.html">plotNMF</a></td>
<td>Plot specific reduced dimensions</td></tr>
<tr><td style="width: 25%;"><a href="plot_reddim.html">plotPCA</a></td>
<td>Plot specific reduced dimensions</td></tr>
<tr><td style="width: 25%;"><a href="plot_reddim.html">plotPCA-method</a></td>
<td>Plot specific reduced dimensions</td></tr>
<tr><td style="width: 25%;"><a href="plot_reddim.html">plotPCASCE</a></td>
<td>Plot specific reduced dimensions</td></tr>
<tr><td style="width: 25%;"><a href="plotPlatePosition.html">plotPlatePosition</a></td>
<td>Plot cells in plate positions</td></tr>
<tr><td style="width: 25%;"><a href="plotReducedDim.html">plotReducedDim</a></td>
<td>Plot reduced dimensions</td></tr>
<tr><td style="width: 25%;"><a href="plotRLE.html">plotRLE</a></td>
<td>Plot relative log expression</td></tr>
<tr><td style="width: 25%;"><a href="plotRLE.html">plotRLE-method</a></td>
<td>Plot relative log expression</td></tr>
<tr><td style="width: 25%;"><a href="plotRowData.html">plotRowData</a></td>
<td>Plot row metadata</td></tr>
<tr><td style="width: 25%;"><a href="plotScater.html">plotScater</a></td>
<td>Plot an overview of expression for each cell</td></tr>
<tr><td style="width: 25%;"><a href="plot_reddim.html">plotTSNE</a></td>
<td>Plot specific reduced dimensions</td></tr>
<tr><td style="width: 25%;"><a href="plot_reddim.html">plotUMAP</a></td>
<td>Plot specific reduced dimensions</td></tr>
<tr><td style="width: 25%;"><a href="projectReducedDim.html">projectReducedDim</a></td>
<td>Project cells into an arbitrary dimensionality reduction space.</td></tr>
<tr><td style="width: 25%;"><a href="projectReducedDim.html">projectReducedDim-method</a></td>
<td>Project cells into an arbitrary dimensionality reduction space.</td></tr>
<tr><td style="width: 25%;"><a href="plot_reddim.html">Reduced dimension plots</a></td>
<td>Plot specific reduced dimensions</td></tr>
<tr><td style="width: 25%;"><a href="retrieveCellInfo.html">retrieveCellInfo</a></td>
<td>Cell-based data retrieval</td></tr>
<tr><td style="width: 25%;"><a href="retrieveFeatureInfo.html">retrieveFeatureInfo</a></td>
<td>Feature-based data retrieval</td></tr>
<tr><td style="width: 25%;"><a href="runColDataPCA.html">runColDataPCA</a></td>
<td>Perform PCA on column metadata</td></tr>
<tr><td style="width: 25%;"><a href="defunct.html">runDiffusionMap</a></td>
<td>Defunct functions</td></tr>
<tr><td style="width: 25%;"><a href="runMDS.html">runMDS</a></td>
<td>Perform MDS on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="runMultiUMAP.html">runMultiUMAP</a></td>
<td>Multi-modal UMAP</td></tr>
<tr><td style="width: 25%;"><a href="runNMF.html">runNMF</a></td>
<td>Perform NMF on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="runPCA.html">runPCA</a></td>
<td>Perform PCA on expression data</td></tr>
<tr><td style="width: 25%;"><a href="runPCA.html">runPCA-method</a></td>
<td>Perform PCA on expression data</td></tr>
<tr><td style="width: 25%;"><a href="runTSNE.html">runTSNE</a></td>
<td>Perform t-SNE on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="runUMAP.html">runUMAP</a></td>
<td>Perform UMAP on cell-level data</td></tr>
<tr><td style="width: 25%;"><a href="scater-pkg.html">scater-pkg</a></td>
<td>The 'scater' package</td></tr>
<tr><td style="width: 25%;"><a href="scater-plot-args.html">scater-plot-args</a></td>
<td>General visualization parameters</td></tr>
<tr><td style="width: 25%;"><a href="SCESet.html">SCESet</a></td>
<td>The "Single Cell Expression Set" (SCESet) class</td></tr>
<tr><td style="width: 25%;"><a href="SCESet.html">SCESet-class</a></td>
<td>The "Single Cell Expression Set" (SCESet) class</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">stand_exprs</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">stand_exprs,SingleCellExperiment-method,</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">stand_exprs&lt;-</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="accessors.html">stand_exprs&lt;--method</a></td>
<td>Additional accessors for the typical elements of a SingleCellExperiment object.</td></tr>
<tr><td style="width: 25%;"><a href="toSingleCellExperiment.html">toSingleCellExperiment</a></td>
<td>Convert an SCESet object to a SingleCellExperiment object</td></tr>
<tr><td style="width: 25%;"><a href="toSingleCellExperiment.html">updateSCESet</a></td>
<td>Convert an SCESet object to a SingleCellExperiment object</td></tr>
</table>
</div></body></html>
